src/wasm/wasm-type.cpp

```diff
@@ -2052,6 +2082,11 @@ bool FiniteShapeEquator::eq(const TypeInfo& a, const TypeInfo& b) {
 }
 
 bool FiniteShapeEquator::eq(const HeapTypeInfo& a, const HeapTypeInfo& b) {
+  if (a.isNominal != b.isNominal) {
+    return false;
+  } else if (a.isNominal) {
+    return &a == &b;
+  }
```

Comment:
To be sure I follow, there is no old code for typesystem==nominal here because we never even got here in the first place?

Reply:
That's right. `FiniteShapeEquator` and `FiniteShapeHasher` are only used as part of equirecursive canonicalization, so they're skipped entirely in `--nominal` mode.